src/portability.rs

```diff
@@ -17,12 +18,13 @@ pub fn hamming(input_num: u64) -> u32 {
 }
 
 #[cfg_attr(not(feature = "no-inline"), inline)]
-#[cfg(target_arch = "x86")]
+#[cfg(target_arch = "x86_64")]
 pub fn hamming(input_num: u64) -> u32 {
     unsafe { __popcnt(input_num as u32) + __popcnt((input_num >> 32) as u32) as u32 }
 }
 
 #[cfg_attr(not(feature = "no-inline"), inline)]
+#[cfg(target_arch = "x86_64")]
```

Comment:
This should guard for both x86 and x86_64 architectures, right now it'd not compile on a 32 bit system.